crates/tako/src/scheduler/state.rs

```diff
@@ -188,23 +206,53 @@ impl SchedulerState {
             );
         }
 
+        let (task_map, worker_map) = core.split_tasks_workers_mut();
         for (worker_id, mut task_ids) in task_computes {
             task_ids.sort_by_cached_key(|&task_id| {
-                let task = core.get_task(task_id);
+                let task = task_map.get_task(task_id);
                 Reverse((task.configuration.user_priority, task.scheduler_priority))
             });
-            let tasks = core.task_map_mut();
             for task_id in task_ids {
-                {
-                    let task = tasks.get_task_mut(task_id);
-                    task.set_fresh_flag(false);
+                let task = task_map.get_task_mut(task_id);
+                task.set_fresh_flag(false);
+                comm.send_worker_message(worker_id, &task.make_compute_message(Vec::new()));
+            }
+        }
+
+        // Try unreserve workers
+        for worker in worker_map.get_workers_mut() {
+            if worker.is_reserved() {
+                let unreserve = worker
+                    .mn_task()
+                    .map(|mn| {
+                        let task = task_map.get_task(mn.task_id);
+                        task.mn_root_worker().unwrap() == worker.id
+                    })
+                    .unwrap_or(true);
+                if unreserve {
+                    worker.set_reservation(false);
```

Comment:
It would be nice to create a function that calls set_reservation + sends the change to the worker, so that we don't forget (it's in multiple places).